### PR TITLE
tikv-control: support the Flashback command trigger

### PR DESCRIPTION
**Pull request by HuSharp**
### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions (in Chinese)](https://github.com/pingcap/docs-cn/blob/master/CONTRIBUTING.md#版本选择指南).

- [x] v6.5 (TiDB 6.5 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s): 
TiKV: https://github.com/tikv/tikv/pull/14875  
PD: https://github.com/tikv/pd/pull/6558 
metrics https://github.com/tikv/tikv/pull/14843

### CODE REVIEW

File: pd-control.md [hunk 1]
```diff
@@ -452,6 +452,8 @@ config show cluster-version
 
 - `store-limit-mode` 用于控制 store 限速机制的模式。主要有两种模式：`auto` 和 `manual`。`auto` 模式下会根据 load 自动进行平衡调整（实验性功能）。
 
+- `halt-scheduling` 用于停止调度。开启该配置后，PD 将停止调度，并且忽略其他的调度配置。
```
Comment (ran-huang) [inline pd-control.md:455]: 这是 6.5.3 的新配置吗？

Comment (HuSharp) [reply pd-control.md:455]: 是的 pd cp 代码贴到了上面的链接中

File: tikv-control.md [hunk 2]
```diff
@@ -548,6 +548,36 @@ success!
 > - `-p` 选项指定 PD 的 endpoint，不使用 `http` 前缀，用于查询指定的 `region_id` 是否有效。
 > - 对于指定 Region 的 peers 所在的每个 store，均须执行该命令。
 
+### Flashback
```
Comment (ran-huang) [inline tikv-control.md:551]: 这是 tikv-ctl 6.5.3 的新功能吗？

Comment (HuSharp) [reply tikv-control.md:551]: 是的 tikv 的 cp 也贴到了 pr 描述中

File: pd-control.md (hunk 1 above)
Comment (ran-huang) [reply pd-control.md:455]: 这两处都请标注版本吧，比如“从 v6.5.3 起引入”

File: tikv-control.md [hunk 3]
```diff
@@ -548,6 +548,36 @@ success!
...
+
+TiDB v6.4.0 引入了 [`FLASHBACK CLUSTER TO TIMESTAMP`](/sql-statements/sql-statement-flashback-to-timestamp.md) 语法，其功能是将集群的数据恢复到特定的时间点。为了方便脱离 TiDB 使用，tikv-ctl 自 v6.5.3 开始提供了 `flashback` 命令。该命令支持在 TiKV 层面进行 Flashback 操作。
+
+> **注意：**
+>
+> - 在执行 `flashback` 命令前，需要先通过 `./pd-ctl config set halt-scheduling true` 命令[停止 PD 调度](pd-control.md#config-show--set-option-value--placement-rules)。
+> - 该命令是用最新的时间戳写入特定时间点的旧数据，但不会删除当前数据，所以在使用前请确保集群有足够的存储空间来同时容纳旧数据和当前数据。
```
Comment (ran-huang) [inline tikv-control.md:558]:
```suggestion
> - `flashback` 命令使用最新的时间戳写入特定时间点的旧数据，但不会删除当前数据，所以在使用前请确保集群有足够的存储空间来同时容纳旧数据和当前数据。
```

File: tikv-control.md [hunk 4]
```diff
@@ -548,6 +548,36 @@ success!
...
+
+```
+flashback all stores success!
+```
+
+上述命令中各选项的含义如下：
+
+- `--pd` 用于指定 PD 的访问地址。
+- `-v` 用于指定 Flashback 目标的时间点。
+- 默认情况下，该命令会对整个集群进行 Flashback。如果需要对指定 Region 或 key 范围进行操作，可以：
```
Comment (ran-huang) [inline tikv-control.md:577]:
```suggestion
- 默认情况下，该命令会对整个集群进行 Flashback。如果需要对指定 Region 或 key 范围进行操作，可以使用以下选项：
```

File: tikv-control.md [hunk 5]
```diff
@@ -548,6 +548,36 @@ success!
...
+    - 使用 `-r` 选项指定 Region，多个 Region 之间用 `,` 分隔。
+    - 使用 `--start` 和 `--end` 指定某个 key 范围内的所有 Region（默认无范围限制，采用 Hex 格式）。
```
Comment (ran-huang) [inline tikv-control.md:579]: 这两个选项能否也提供一个示例？

Comment (HuSharp) [reply tikv-control.md:580]: 已更新

File: tikv-control.md [hunk 6]
```diff
@@ -548,6 +548,61 @@ success!
...
+### Flashback
+
+TiDB v6.4.0 引入了 [`FLASHBACK CLUSTER TO TIMESTAMP`](/sql-statements/sql-statement-flashback-to-timestamp.md) 语法，其功能是将集群的数据恢复到特定的时间点。为了方便脱离 TiDB 使用，tikv-ctl 自 v6.5.3 开始提供了 `flashback` 命令。该命令支持在 TiKV 层面进行 Flashback 操作。
+
+> **注意：**
+>
+> - 在执行 `flashback` 命令前，需要先通过 `./pd-ctl config set halt-scheduling true` 命令[停止 PD 调度](pd-control.md#config-show--set-option-value--placement-rules)。
+> - `flashback` 命令使用最新的时间戳写入特定时间点的旧数据，但不会删除当前数据，所以在使用前请确保集群有足够的存储空间来同时容纳旧数据和当前数据。
+> - 该命令只支持本地模式。运行成功后，会打印 `flashback all stores success!`。
+> - 可以通过 [Peer in Flashback State](./grafana-tikv-dashboard.md#raft-admin) 查看执行进度。
```
Comment (Oreoxmt) [inline tikv-control.md:560]:
```suggestion
> - 可以通过 [Raft admin > Peer in Flashback State](/grafana-tikv-dashboard.md#raft-admin) 监控项查看执行进度。
```

File: tikv-control.md [hunk 7]
```diff
@@ -548,6 +548,61 @@ success!
...
+### Flashback
+
+TiDB v6.4.0 引入了 [`FLASHBACK CLUSTER TO TIMESTAMP`](/sql-statements/sql-statement-flashback-to-timestamp.md) 语法，其功能是将集群的数据恢复到特定的时间点。为了方便脱离 TiDB 使用，tikv-ctl 自 v6.5.3 开始提供了 `flashback` 命令。该命令支持在 TiKV 层面进行 Flashback 操作。
+
+> **注意：**
+>
+> - 在执行 `flashback` 命令前，需要先通过 `./pd-ctl config set halt-scheduling true` 命令[停止 PD 调度](pd-control.md#config-show--set-option-value--placement-rules)。
+> - `flashback` 命令使用最新的时间戳写入特定时间点的旧数据，但不会删除当前数据，所以在使用前请确保集群有足够的存储空间来同时容纳旧数据和当前数据。
+> - 该命令只支持本地模式。运行成功后，会打印 `flashback all stores success!`。
+> - 可以通过 [Raft admin > Peer in Flashback State](/grafana-tikv-dashboard.md#raft-admin) 监控项查看执行进度。
```
Comment (ran-huang) [inline tikv-control.md:560]:
```suggestion
> **注意：**
>
> - `flashback` 命令使用最新的时间戳写入特定时间点的旧数据，但不会删除当前数据，所以在使用前请确保集群有足够的存储空间来同时容纳旧数据和当前数据。
> - `flashback` 命令只支持本地模式。
```

File: tikv-control.md [hunk 8]
```diff
@@ -548,6 +548,61 @@ success!
...
+flashback all stores success!
+```
+
+上述命令中各选项的含义如下：
+
+- `--pd` 用于指定 PD 的访问地址。
+- `-v` 用于指定 Flashback 目标的时间点。
+- 默认情况下，该命令会对整个集群进行 Flashback。如果需要对指定 Region 或 key 范围进行操作，可以使用以下选项：
+    - 使用 `-r` 选项指定 Region，多个 Region 之间用 `,` 分隔。
+    - 使用 `--start` 和 `--end` 指定某个 key 范围内的所有 Region（默认无范围限制，采用 Hex 格式）。
```
Comment (ran-huang) [inline tikv-control.md:580]:
````suggestion
#### 使用方式

```shell
tikv-ctl --pd <pd_address:port> flashback -v <target_timestamp>
```

使用 `--pd` 选项指定 PD 的访问地址。使用 `-v` 选项指定 Flashback 目标的时间点。

默认情况下，该命令会对整个集群进行 Flashback。如果需要对指定 Region 或 key 范围进行操作，可以使用以下选项：

- 使用 `-r` 选项指定 Region，多个 Region 之间用 `,` 分隔。
- 使用 `--start` 和 `--end` 指定某个 key 范围内的所有 Region（默认无范围限制，采用 Hex 格式）。

命令运行成功后，会打印 `flashback all stores success!`。你也可以通过 [Raft admin > Peer in Flashback State](/grafana-tikv-dashboard.md#raft-admin) 监控项查看执行进度。
````

File: tikv-control.md [hunk 9]
```diff
@@ -548,6 +548,61 @@ success!
...
+
+```shell
+tikv-ctl --pd 127.0.0.1:2379 flashback -v 430315739761082369 --start 7480000000000000FF0800000000000000F8 --end 7480000000000000FF0C00000000000000F8
+```
+
+输出结果如下：
+
+```
+flashback all stores success!
+```
```
Comment (ran-huang) [inline tikv-control.md:604]:
````suggestion
## 示例

下面示例将整个集群的数据恢复到时间点 `430315739761082369`：

```shell
tikv-ctl --pd 127.0.0.1:2379 flashback -v 430315739761082369
```

需要将 ID 为 `100` 和 `102` 的 Region 的数据恢复到时间点 `430315739761082369` 时，使用以下命令：

```shell
tikv-ctl --pd 127.0.0.1:2379 flashback -v 430315739761082369 -r 100,102
```

需要将 key 范围的数据恢复到时间点 `430315739761082369` 时，使用以下命令：

```shell
tikv-ctl --pd 127.0.0.1:2379 flashback -v 430315739761082369 --start 7480000000000000FF0800000000000000F8 --end 7480000000000000FF0C00000000000000F8
```
````

File: tikv-control.md [hunk 10]
```diff
@@ -548,6 +548,61 @@ success!
...
+
+TiDB v6.4.0 引入了 [`FLASHBACK CLUSTER TO TIMESTAMP`](/sql-statements/sql-statement-flashback-to-timestamp.md) 语法，其功能是将集群的数据恢复到特定的时间点。为了方便脱离 TiDB 使用，tikv-ctl 自 v6.5.3 开始提供了 `flashback` 命令。该命令支持在 TiKV 层面进行 Flashback 操作。
+
+> **注意：**
+>
+> - 在执行 `flashback` 命令前，需要先通过 `./pd-ctl config set halt-scheduling true` 命令[停止 PD 调度](pd-control.md#config-show--set-option-value--placement-rules)。
+> - `flashback` 命令使用最新的时间戳写入特定时间点的旧数据，但不会删除当前数据，所以在使用前请确保集群有足够的存储空间来同时容纳旧数据和当前数据。
+> - 该命令只支持本地模式。运行成功后，会打印 `flashback all stores success!`。
+> - 可以通过 [Raft admin > Peer in Flashback State](/grafana-tikv-dashboard.md#raft-admin) 监控项查看执行进度。
+
```
Comment (ran-huang) [inline tikv-control.md:561]:
```suggestion

#### 前置条件

在执行 `flashback` 命令前，需要先通过 `./pd-ctl config set halt-scheduling true` 命令[停止 PD 调度](pd-control.md#config-show--set-option-value--placement-rules)。

```

**Review by ran-huang — APPROVED** [review]

File: tikv-control.md [hunk 11]
```diff
@@ -548,6 +548,54 @@ success!
...
+```shell
+tikv-ctl --pd 127.0.0.1:2379 flashback -v 430315739761082369 -r 100,102
+```
+
+需要将 key 范围的数据恢复到时间点 `430315739761082369` 时，使用以下命令：
+
+```shell
+tikv-ctl --pd 127.0.0.1:2379 flashback -v 430315739761082369 --start 7480000000000000FF0800000000000000F8 --end 7480000000000000FF0C00000000000000F8
+```
```
Comment (Oreoxmt) [inline tikv-control.md:597]:
````suggestion
- 下面示例将整个集群的数据恢复到时间点 `430315739761082369`：

    ```shell
    tikv-ctl --pd 127.0.0.1:2379 flashback -v 430315739761082369
    ```

- 需要将 ID 为 `100` 和 `102` 的 Region 的数据恢复到时间点 `430315739761082369` 时，使用以下命令：

    ```shell
    tikv-ctl --pd 127.0.0.1:2379 flashback -v 430315739761082369 -r 100,102
    ```

- 需要将 key 范围的数据恢复到时间点 `430315739761082369` 时，使用以下命令：

    ```shell
    tikv-ctl --pd 127.0.0.1:2379 flashback -v 430315739761082369 --start 7480000000000000FF0800000000000000F8 --end 7480000000000000FF0C00000000000000F8
   ```
````

**Review by Connor1996 — APPROVED** [review]
LGTM

**Pull request merged**